runtime/altair/src/migrations.rs

```diff
@@ -9,5 +9,163 @@
 // but WITHOUT ANY WARRANTY; without even the implied warranty of
 // MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 // GNU General Public License for more details.
+use cfg_primitives::Balance;
+use cfg_types::tokens::CurrencyId;
+use frame_support::{traits::OnRuntimeUpgrade, weights::Weight};
 
-pub type UpgradeAltair1028 = ();
+use crate::Runtime;
+
+pub type UpgradeAltair1028 = (asset_registry::CrossChainTransferabilityMigration,);
+
+mod asset_registry {
+	use cfg_types::{tokens as v1, tokens::CustomMetadata};
+	#[cfg(feature = "try-runtime")]
+	use frame_support::ensure;
+	use frame_support::{pallet_prelude::OptionQuery, storage_alias, Twox64Concat};
+	use orml_traits::asset_registry::AssetMetadata;
+	#[cfg(feature = "try-runtime")]
+	use sp_std::vec::Vec;
+
+	use super::*;
+	use crate::VERSION;
+
+	/// Migrate all the registered asset's metadata to the new version of
+	/// `CustomMetadata` which contains a `CrossChainTransferability` property.
+	/// At this point in time, the `transferability` of Tranche tokens should be
+	/// set to `CrossChainTransferability::Xcm` and for all other tokens to
+	/// `CrossChainTransferability::Xcm`, with the exception of
+	/// `Currency::Staking` tokens which are not registered in the first place.
+	pub struct CrossChainTransferabilityMigration;
+
+	// The old orml_asset_registry Metadata storage using v0::CustomMetadata
+	#[storage_alias]
+	type Metadata<T: orml_asset_registry::Config> = StorageMap<
+		orml_asset_registry::Pallet<T>,
+		Twox64Concat,
+		CurrencyId,
+		AssetMetadata<Balance, v0::CustomMetadata>,
+		OptionQuery,
+	>;
+
+	impl OnRuntimeUpgrade for CrossChainTransferabilityMigration {
+		fn on_runtime_upgrade() -> Weight {
+			// todo(nuno): this migration will **most likely** be part of the set of
+			// migrations when upgrading to Centrifuge 1020. Refine when we know
+			// for sure.
+			if VERSION.spec_version != 1020 {
```

Comment:
I guess this would be updated accordingly to the Altair spec version once we know the final version

Reply:
Ah, good catch! I copied this over from the Centrifuge runtime, we have to adjust this

Reply:
Done, thanks again!